audbcards/core/datacard.py

```diff
@@ -161,83 +147,143 @@ def file_duration_distribution(self) -> str:
+                audeer.mkdir(os.path.dirname(cache_file))
+                self._plot_distribution(durations)
+                plt.savefig(cache_file, transparent=True)
+                plt.close()
+
             image_file = audeer.path(
                 self.sphinx_src_dir,
                 self.path,
                 self.dataset.name,
-                f"{self.dataset.name}-{name}.png",
+                file_name,
             )
             audeer.mkdir(os.path.dirname(image_file))
-            plt.savefig(image_file, transparent=True)
-            plt.close()
+            shutil.copyfile(cache_file, image_file)
             distribution_str = self._inline_image(
                 f"{min_:.1f} {unit}",
-                f"./{self.dataset.name}/{self.dataset.name}-{name}.png",
+                f"./{self.dataset.name}/{file_name}",
                 f"{max_:.1f} {unit}",
             )
 
         return distribution_str
 
-    def player(
-        self,
-        file: str = None,
-    ) -> str:
+    def player(self) -> str:
         r"""Create an audio player showing the waveform.
 
-        Args:
-            file: input audio file to be used in the player.
-                If ``None``,
-                :attr:`audbcards.Datacard.example_media`
-                is used
+        If :attr:`audbcards.Datacard.sphinx_build_dir`
+        or :attr:`audbcards.Datacard.sphinx_src_dir`
+        is not ``None``,
+        an example media file is cached in the folder
+        ``<dataset-name>-<dataset-version>-player-media/``
+        inside
+        ``<cache-root>/<dataset-name>/<dataset-version>/``,
+        using the same sub-folder structure
+        as the media file has inside its dataset.
+        If :attr:`audbcards.Datacard.sphinx_build_dir`
+        is not ``None``,
+        the media sub-folder structure
+        is also copied
+        to the sphinx build dir into
+        ``<sphinx-build-dir>/<path>/<dataset-name>/``.
+
+        If :attr:`audbcards.Datacard.sphinx_src_dir` is not ``None``,
+        a plot of the waveform of the media file
+        is cached under
+        ``<dataset-name>-<dataset-version>-player-waveform.png``
+        inside
+        ``<cache-root>/<dataset-name>/<dataset-version>/``.
+        It is also copied to the sphinx source folder into
+        ``<sphinx-src-dir>/<path>/<dataset-name>/``.
 
-        """
-        if file is None:
-            file = self.example_media
+        Returns:
+            String containing RST code to include the player
 
-        # use audb cache instead of dataset.cache_root
-        media_src_dir = (
-            f"{audb.default_cache_root()}/"
-            f"{audb.flavor_path(self.dataset.name, self.dataset.version)}"
+        """
+        # Cache is organized as `<cache_root>/<name>/<version>/`
+        cache_folder = audeer.path(
+            self.cache_root,
+            self.dataset.name,
+            self.dataset.version,
+        )
+        cache_example_media = audeer.path(
```

Comment:
`cache_example_media` is not used before line 240. Would it make sense to define it closer to where is is actually used?

Reply:
I moved the definition of `cache_example_media` into `load_media_to_cache()`.

Reply:
There has been quite some refactoring going on. The clause after

```
if self.sphinx_src_dir is not None:
```
is now neat and clean. I am closing the other conversations / threads about this refactoring without specific commenting.